mat64/io.go

```diff
@@ -81,18 +87,24 @@ func (m *Dense) UnmarshalBinary(data []byte) error {
 		panic("mat64: unmarshal into non-zero matrix")
 	}
 
+	var tmp [8]byte
 	buf := bytes.NewReader(data)
-	var rows int64
-	err := binary.Read(buf, binary.LittleEndian, &rows)
+	_, err := buf.Read(tmp[:])
```

Comment:
good call.

the performances are a bit deteriorated wrt the sans-ReadFull:

```
name                    old time/op    new time/op    delta
MarshalDense10-4           668ns ± 2%     691ns ± 4%    +3.47%          (p=0.048 n=5+5)
MarshalDense100-4         3.99µs ± 1%    4.07µs ± 2%      ~             (p=0.095 n=5+5)
MarshalDense1000-4        36.3µs ± 1%    36.8µs ± 2%      ~             (p=0.222 n=5+5)
MarshalDense10000-4        349µs ± 1%     349µs ± 1%      ~             (p=0.841 n=5+5)
UnmarshalDense10-4         425ns ± 1%     778ns ± 3%   +83.10%          (p=0.008 n=5+5)
UnmarshalDense100-4       3.11µs ± 1%    5.13µs ± 2%   +64.72%          (p=0.008 n=5+5)
UnmarshalDense1000-4      29.2µs ± 1%    48.4µs ± 1%   +65.86%          (p=0.008 n=5+5)
UnmarshalDense10000-4      281µs ± 2%     464µs ± 4%   +65.08%          (p=0.008 n=5+5)
MarshalVector10-4          638ns ± 1%     642ns ± 7%      ~             (p=0.548 n=5+5)
MarshalVector100-4        3.89µs ± 1%    3.94µs ± 4%      ~             (p=0.690 n=5+5)
MarshalVector1000-4       36.0µs ± 1%    36.5µs ± 5%      ~             (p=0.690 n=5+5)
MarshalVector10000-4       348µs ± 1%     351µs ± 4%      ~             (p=0.841 n=5+5)
UnmarshalVector10-4        411ns ± 1%     733ns ± 1%   +78.39%          (p=0.008 n=5+5)
UnmarshalVector100-4      3.13µs ± 4%    5.17µs ± 2%   +65.50%          (p=0.008 n=5+5)
UnmarshalVector1000-4     29.6µs ± 3%    49.0µs ± 1%   +65.43%          (p=0.008 n=5+5)
UnmarshalVector10000-4     290µs ± 2%     477µs ± 7%   +64.39%          (p=0.008 n=5+5)

name                    old alloc/op   new alloc/op   delta
MarshalDense10-4            208B ± 0%      208B ± 0%      ~     (all samples are equal)
MarshalDense100-4         1.01kB ± 0%    1.01kB ± 0%      ~     (all samples are equal)
MarshalDense1000-4        8.30kB ± 0%    8.30kB ± 0%      ~     (all samples are equal)
MarshalDense10000-4       82.0kB ± 0%    82.0kB ± 0%      ~     (all samples are equal)
UnmarshalDense10-4         80.0B ± 0%    136.0B ± 0%   +70.00%          (p=0.008 n=5+5)
UnmarshalDense100-4         896B ± 0%      952B ± 0%    +6.25%          (p=0.008 n=5+5)
UnmarshalDense1000-4      8.19kB ± 0%    8.25kB ± 0%    +0.68%          (p=0.008 n=5+5)
UnmarshalDense10000-4     81.9kB ± 0%    82.0kB ± 0%    +0.07%          (p=0.016 n=4+5)
MarshalVector10-4           208B ± 0%      208B ± 0%      ~     (all samples are equal)
MarshalVector100-4        1.01kB ± 0%    1.01kB ± 0%      ~     (all samples are equal)
MarshalVector1000-4       8.30kB ± 0%    8.30kB ± 0%      ~     (all samples are equal)
MarshalVector10000-4      82.0kB ± 0%    82.0kB ± 0%      ~     (all samples are equal)
UnmarshalVector10-4        80.0B ± 0%    136.0B ± 0%   +70.00%          (p=0.008 n=5+5)
UnmarshalVector100-4        896B ± 0%      952B ± 0%    +6.25%          (p=0.008 n=5+5)
UnmarshalVector1000-4     8.19kB ± 0%    8.25kB ± 0%    +0.68%          (p=0.008 n=5+5)
UnmarshalVector10000-4    81.9kB ± 0%    82.0kB ± 0%    +0.07%          (p=0.008 n=5+5)

name                    old allocs/op  new allocs/op  delta
MarshalDense10-4            2.00 ± 0%      2.00 ± 0%      ~     (all samples are equal)
MarshalDense100-4           2.00 ± 0%      2.00 ± 0%      ~     (all samples are equal)
MarshalDense1000-4          2.00 ± 0%      2.00 ± 0%      ~     (all samples are equal)
MarshalDense10000-4         2.00 ± 0%      2.00 ± 0%      ~     (all samples are equal)
UnmarshalDense10-4          1.00 ± 0%      3.00 ± 0%  +200.00%          (p=0.008 n=5+5)
UnmarshalDense100-4         1.00 ± 0%      3.00 ± 0%  +200.00%          (p=0.008 n=5+5)
UnmarshalDense1000-4        1.00 ± 0%      3.00 ± 0%  +200.00%          (p=0.008 n=5+5)
UnmarshalDense10000-4       1.00 ± 0%      3.00 ± 0%  +200.00%          (p=0.008 n=5+5)
MarshalVector10-4           2.00 ± 0%      2.00 ± 0%      ~     (all samples are equal)
MarshalVector100-4          2.00 ± 0%      2.00 ± 0%      ~     (all samples are equal)
MarshalVector1000-4         2.00 ± 0%      2.00 ± 0%      ~     (all samples are equal)
MarshalVector10000-4        2.00 ± 0%      2.00 ± 0%      ~     (all samples are equal)
UnmarshalVector10-4         1.00 ± 0%      3.00 ± 0%  +200.00%          (p=0.008 n=5+5)
UnmarshalVector100-4        1.00 ± 0%      3.00 ± 0%  +200.00%          (p=0.008 n=5+5)
UnmarshalVector1000-4       1.00 ± 0%      3.00 ± 0%  +200.00%          (p=0.008 n=5+5)
UnmarshalVector10000-4      1.00 ± 0%      3.00 ± 0%  +200.00%          (p=0.008 n=5+5)
```

I suppose we can live with it though.